code/modules/antagonists/_common/antag_exfiltration.dm

```diff
@@ -0,0 +1,600 @@
+// Traitor flare
+/obj/item/wormhole_jaunter/extraction
+	name = "extraction flare"
+	icon = 'icons/obj/lighting.dmi'
+	desc = "A single-use extraction flare that will let you escape the station. One way trip."
+	icon_state = "flare-contractor"
+	item_state = "flare"
+	/// What areas can be extracted from?
+	var/list/extractable_areas = list()
+	/// Is there a roundstart delay?
+	var/delayed_extraction = TRUE
+	/// How long does the extraction take?
+	var/extraction_time = 90 SECONDS
+	/// Type of setup once a flare is lit
+	var/setup_type = /obj/effect/temp_visual/exfiltration
+	/// Type of portal spawned
+	var/portal_type = /obj/effect/portal/advanced/exfiltration
+
+/obj/item/wormhole_jaunter/extraction/Initialize(mapload)
+	. = ..()
+	var/list/possible_areas = list(
```

Comment:
Why a list of string names of the areas, as opposed to a whitelist of `/area/station` substypes? Relying directly on the type paths would also probably mean you don't have to do the checks for indoors, on-station-level, areas.